conans/client/migrations.py

```diff
@@ -132,6 +143,43 @@ def _make_migrations(self, old_version):
                 migrate_c_src_export_source(self.cache, self.out)
 
 
+def _get_refs(cache):
+    folders = list_folder_subdirs(cache.store, 4)
+    return [ConanFileReference(*s.split("/")) for s in folders]
+
+
+def _get_prefs(cache, ref):
+    packages_folder = cache.packages(ref)
+    folders = list_folder_subdirs(packages_folder, 1)
+    return [PackageReference(ref, s) for s in folders]
+
+
+def _migrate_create_metadata(cache, out):
+    out.warn("Migration: Generating missing metadata files")
+    refs = _get_refs(cache)
+
+    for ref in refs:
+        try:
+            folder = cache.export(ref)
+            manifest = FileTreeManifest.load(folder)
+            metadata_path = os.path.join(cache.conan(ref), PACKAGE_METADATA)
+            if not os.path.exists(metadata_path):
+                out.info("Creating {} for {}".format(PACKAGE_METADATA, ref))
+                prefs = _get_prefs(cache, ref)
+                metadata = PackageMetadata()
+                metadata.recipe.revision = manifest.summary_hash
```

Comment:
Maybe this revision could have a proper name, as it will have nothing to do with the following ones (those will be the commit hash). A name saying that is was generated from an old Conan version at a certain timestamp.

Reply:
It is the regular revision when you don't have `scm` or when, in the future, you can choose if hash or not hash. What do you want exactly to name here? This is just to create a default. The hash as the revision is a good start for old packages.

Reply:
If all the revisions but this one will be named after the `scm` sha (I choose to use the default behavior with a git repo) then I will end up with a list of revisions corresponding to commits in my repository; I can clone any of them but this one, and it looks like a commit sha but it isn't... If we name it "Conan migration - <timestamp>" I'm sure it is not a commit and I won't try to clone it (and wonder why I lost that commit in my repo).

Reply:
tiny improvement...